qiskit_experiments/analysis/curve_analysis.py

```diff
@@ -231,9 +231,24 @@ def __new__(cls) -> "CurveAnalysis":
     def __init__(self):
         """Initialize data fields that are privately accessed by methods."""
 
+        # str: Experiment type
+        self.__experiment_type = None
```

Comment:
Is there a particular reason to use name mangling for all these variables?

Reply:
This meant to say these values are not expected to be overridden during the analysis and consumed internally (thus I only give protected getter functions, no setter, no public). If they look bit complicated, I'll drop these getters and define them as protected instance values.